### PR TITLE
introduce TcpSocketHostLocator extension point

### PR DESCRIPTION
**Pull request by ndeloof**
relies on https://github.com/jenkinsci/maven-interceptors/pull/8
refactor MAVEN_REMOTE_USEINET so this one is just one-of case communication between maven agent and slave agent do require some netork tweaks

### CODE REVIEW

**Comment by ndeloof** [comment]
@reviewbybees

File: src/main/java/hudson/maven/TcpSocketHostLocator.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    /**
+     * Try to identify the slave agent TCP socket host name or IP.
+     * @return <code>null</code> if not found or does not apply to this specific implementation
+     * @throws IOException
+     */
+    public abstract @CheckForNull String getTcpSocketHost() throws IOException;
+
+    public static List<TcpSocketHostLocator> all() {
+        return Jenkins.getInstance().getExtensionList(TcpSocketHostLocator.class);
```
Comment (jglick) [inline src/main/java/hudson/maven/TcpSocketHostLocator.java]: `ExtensionList.lookup`

File: src/main/java/hudson/maven/TcpSocketHostLocator.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * Identify the Host name to use from maven-agent to connect to to jenkins slave agent TCP socket.
+ * <p>
+ * In simple scenarios both slave agent and maven process do live on same host without specific network
+ * constraints, but for some virtualization usages maven process just can't bind a socket on wildcard
+ * host network. This extension give infrastructure plugins a chance to configure the adequate hostname.
+ * to handle such network constraints
+ *
+ * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
+ */
+public abstract class TcpSocketHostLocator {
```
Comment (jglick) [inline src/main/java/hudson/maven/TcpSocketHostLocator.java]: `implements ExtensionPoint`

File: pom.xml [hunk 3]
```diff
@@ -43,7 +43,7 @@ THE SOFTWARE.
   <url>http://wiki.jenkins-ci.org/display/JENKINS/Maven+Project+Plugin</url>
 
   <properties>
-    <mavenInterceptorsVersion>1.6</mavenInterceptorsVersion>
+    <mavenInterceptorsVersion>1.7-SNAPSHOT</mavenInterceptorsVersion>
```
Comment (jglick) [inline pom.xml]: Is this relying on a particular commit or PR in that project? I find it useful to note that right in the PR diff, where it is available to reviewers:

```xml
<mavenInterceptorsVersion>1.7-SNAPSHOT</mavenInterceptorsVersion> <!-- TODO …URL… -->
```

**Comment by jglick** [comment]
You will need to `mvn clean deploy` the upstream PR.

**Comment by aheritier** [comment]
:bee:

**Comment by ndeloof** [comment]
marked as WiP waiting for a maven-interceptor release.

**Pull request merged**